### DISCUSSION TITLE
Custom claim with empty collection

### DISCUSSION THREAD

**Discussion by wimdeblauwe** [keycloak/keycloak#9210] [Q&A]
Hi,

I wrote a custom claim mapper. I'd like the claim to be always present, even if the value is an empty collection. I am currently using this helper method in my mapper:

```
OIDCAttributeMapperHelper.mapClaim(token, mappingModel, myCollection);
```

Where `myCollection` could be empty in some cases. When this is the case now, the claim is not added at all, due to this code in that helper class:
```
        if (attributeValue instanceof Collection) {
            Collection<Object> valueAsList = (Collection<Object>) attributeValue;
            if (valueAsList.isEmpty()) return null;
```

Is there any way I can get the claim to be added also if the collection is empty? I am using Keycloak 15.0.1.

**Comment by svenmaerki** [comment]
Hi @wimdeblauwe 
I know you asked this question a long time ago. I was curious if you had discovered a resolution to your issue, as I had encountered the same issue. I am using Keycloak Version 18.
